connectors/src/connectors/github/temporal/activities.ts

```diff
@@ -1167,6 +1208,13 @@ export async function githubCodeSyncActivity({
           });
         }
 
+        await upsertFolderNode({
+          dataSourceConfig,
+          folderId: d.internalId,
+          parents: [d.internalId, ...d.parents, repoId.toString()],
+          title: d.dirName,
+        });
```

Comment:
We are missing the intermediate "github-code-<repoid>" thing, which does not exist in connectors database, but does exists in d.parents.
I also did not check the issues/discussion structure but i think we also have intermediate virtual folders here

Reply:
(no structure for issues, it's just a big database entry)

Reply:
The new commit handles the "Code", "Issues" and "Discussions" nodes